components/layout/layout_task.rs

```diff
@@ -98,26 +98,34 @@ const DISPLAY_PORT_THRESHOLD_SIZE_FACTOR: i32 = 4;
 ///
 /// This needs to be protected by a mutex so we can do fast RPCs.
 pub struct LayoutTaskData {
+    pub frozen: LayoutTaskFrozen,
+    pub unfrozen: LayoutTaskUnfrozen,
+}
+
+pub struct LayoutTaskFrozen {
+    /// The size of the viewport.
+    pub screen_size: Size2D<Au>,
+
+    /// Performs CSS selector matching and style resolution.
+    pub stylist: Stylist,
+
+    /// The position and size of the visible rect for each layer. We do not build display lists
+    /// for any areas more than `DISPLAY_PORT_SIZE_FACTOR` screens away from this area.
+    pub visible_rects: Arc<HashMap<LayerId, Rect<Au>, DefaultState<FnvHasher>>>,
+}
```

Comment:
What does "Frozen"/"Unfrozen" mean? I'm a bit confused. Is "Frozen" immutable? If so, that would be called "Sync" in Rust.